### ISSUE TITLE
allow the FolderResource.createFolders to receive the site

### ISSUE THREAD

**Issue by jdotcms** [dotCMS/core#18419]
Currently the FolderResource is not able to receive the site - and therefore does not work.  This pull request allows a user to send the siteId and create folders under that site.

**Comment by erickgonzalez** [comment]
PR: https://github.com/dotCMS/core/pull/18420

**Comment by erickgonzalez** [comment]
Needs Work:
**siteName** that does not exists throws 200 should throw 400 (bad request).
A user without permissions throws 200 (but it does not create the folders) should throw 401.
Create IT for each.

Works without issues:
Tested on master now the `createfolders` endpoint receives the `siteName` as a path param, and able to create the folders. 
If the parent folder already exists it only creates the child folder.

Endpoint:
```
{{serverURL}}/api/v1/folder/createfolders/{siteName}
```
Json example:
```
["/activities/test1/test2","/newfolder-test-2/","test1"]
```

**Comment by erickgonzalez** [comment]
PR: https://github.com/dotCMS/core/pull/18522

**Comment by bryanboza** [comment]
We need some work with the responses here: 
- If you send an site that does not exist, need to show a 404 instead a 400 
SC: https://www.screencast.com/t/EjyMxvSfNE

- If you try to create a folder with a restricted name, we have a 500 error, and we are not returning  a error message 
SC: https://www.screencast.com/t/R1OfUuE2H

**Comment by dsilvam** [comment]
@bryanboza 
 > If you send an site that does not exist, need to show a 404 instead a 400
SC: https://www.screencast.com/t/EjyMxvSfNE

400 is fine since your a trying to create a folder with the wrong data 

>If you try to create a folder with a restricted name, we have a 500 error, and we are not returning a error message

400 is expected also since this is invalid data

**Comment by erickgonzalez** [comment]
PR for feedback https://github.com/dotCMS/core/pull/18624

**Comment by bryanboza** [comment]
Fixed, tested on release-5.3.2 // Postgres // FF
